middleware/l12n.js

```diff
@@ -0,0 +1,46 @@
+// l12n middleware to redirect requests for URL paths without locale prefixes
+// to the equivalent URL prefixed with the user's preferred locale.
+//
+// (Named "l12n" and not "i18n" as the latter is already taken by nuxt-i18n.)
+//
+// This duplicates the browser language detection functionality of the nuxt-i18n
+// package but with additional redirection handling for requests to URL paths
+// without the language in them.
+
+export default ({ app, route, redirect, req }) => {
+  // Is there a locale in the URL path already?
+  const routePathLocale = route.path.match(/^\/([a-z]{2})(\/|$)/);
+  if (routePathLocale) {
+    // Store it in the cookie, indicating user's current preference e.g. from
+    // using language selector.
+    app.$cookies.set('i18n_locale_code', routePathLocale[1]);
```

Comment:
this should also check the route matches the known list of locales, otherwise you can set a broken cookie.
for example visit:
http://localhost:3000/rf/about-us
then
http://localhost:3000/explore